site/documentation/content/api/management/device-registry-v1.yaml

```diff
@@ -461,7 +487,59 @@ components:
          description: Allows arbitrary properties as extension to the ones
                       specified by the Hono API.
 
-# Tenant schema
+
+      PagingSortingFilteringOptions:
+         type: object
+         additionalProperties: false
+         properties:
+            "page":
+               $ref: '#/components/schemas/PagingOptions'
+            "filters":
+               $ref: '#/components/schemas/FilterOptions'
+            "sort":
+               $ref: '#/components/schemas/SortingOptions'
+
+      PagingOptions:
+         type: object
+         additionalProperties: false
+         description: options to page the results of bulk queries.
+         properties:
+            "limit":
+               type: integer
+               default: 30
+               description: |
+                  The threshold on the number of results returned in a single API call. If not set, the registry will return up to 30 results.
+                  The upper bound for this value is 200 and it cannot be less than 0.
+            "offset":
+               type: integer
+               default: 0
+               description: |
+                  The offset to use as a starting point to return results. This allows to go through the query results in
+                  a paged manner when the query returns more results than the limit.
+
+      FilterOptions:
+            type: object
+            additionalProperties: true
+            description: Allows arbitrary filters to be applied to bulk queries.
+                         Fields to filter MUST be designated with their JSON Path representation.
+            examples:
+               Filters options:
+                  $ref: '#/components/examples/FilterOptionsExample'
+
+      SortingOptions:
+         type: object
+         additionalProperties: true
+         description: Sorting options to be applied to bulk queries results.
+                     The field to be used for sorting MUST be defined with its JSON Path representation.
+         default: ascending
+         enum:
+            - ascending
+            - descending
+         examples:
+            Sorting options:
+               $ref: '#/components/examples/SortingOptionsExample'
```

Comment:
If I understand the examples correctly, then the property names of this object are supposed to be JSON paths like `status.created` and the property values need to be either `ascending` or `descending`, right?
However, my understanding of this definition is that any instance of SortingOptions will be either the literal String `ascending` (default) or `descending`. The enum property does **not** define the possible values of the *additionalProperties* but the value of the *SortingOptions* object itself. Or am I mistaken? I do not think that it would even be possible to express the enum constraint based on your approach of variable property names because AFAIK you can only define constraints on particular properties or objects. So, FMPOV the SortingOptions would need to be changed to something that looks more like this:
```json
[
  {
      "property": "status.created",
      "direction": "descending"
  }, {
       "property": "last-user",
       "direction": "ascending"
  }
]
```

Reply:
I think you are right. I am still not mastering the openAPI specification :)
Your suggestion fits nicely so I updated the spec with that. Thanks !